### PR TITLE
Addede folder with js-dom practice

### PR DESCRIPTION
**Pull request by ASTkach**
### js-dom

[Demo](https://astkach.github.io/js-dom/) |
[Code base](https://github.com/ASTkach/js-dom)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by ASTkach** [comment]
I have completed self-check and fix the code.

**Comment by ASTkach** [comment]
I have completed self-check and fix the code.

File: submissions/ASTkach/js-dom/js/script.js [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+iconMenu.addEventListener("click", function () {
+    document.body.classList.toggle("_lock");
+    iconMenu.classList.toggle("_opened");
+    listMenu.classList.toggle("_opened");
+});
+
+// track the presence of the "_active" class for navigation elements
+// track the "click" event on them
+let navButton = document.getElementsByClassName("nav__button");
+let activeClass = document.getElementsByClassName("_active");
```
Comment (madmaxWMFU) [inline submissions/ASTkach/js-dom/js/script.js:467]: Also, `const`

File: submissions/ASTkach/js-dom/js/script.js [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+
+const mainBlock = document.querySelector(".main");
+
+// function to generate html code for the current planet
+// (sorry, the sun will also be a planet)
+function getCurrentPlanet(planet) {
+    solarSystem.map((item) => {
+        if (item.id == planet) {
+            const planetTitle = item.title;
+            const planetText = item.text;
```
Comment (madmaxWMFU) [inline submissions/ASTkach/js-dom/js/script.js:346]: Once time you use destruction. Also, you can use it here.
`const { title, text, ..... } = item;`

File: submissions/ASTkach/js-dom/js/script.js [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+            let planetTemplateStart = `<div class="main__content">`;
+
+            let planetTemplateEnd = `</div>`;
+
+            let planetTemplateTextContent = `
+                <h1 class="main__title">${planetTitle}</h1>
+                <p class="main__text">${planetText}</p>
+            `;
+
+            let planetTemplateImage = `<img src="${planetImage}" alt="${planetAltText}" class="main__img" width="450" height="450"/>`;
```
Comment (madmaxWMFU) [inline submissions/ASTkach/js-dom/js/script.js:366]: Set styling in CSS

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Good work. See some comments below.
```javascript
const data [{
  title: "",
  size: "",
  description: []
}];

const [ firstElement ] = data;
const { title, size, description } = firstElement;
const template = `
     <div>
       <label>${title}</lable>
       <span>${size}</span>
       ${description.map(el => `<p>${el}</p>`).join('')}`;
```

File: submissions/ASTkach/js-dom/js/script.js [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+}
+
+const navList = document.querySelector(".nav__list");
+
+// track the "click" event on navigation elements
+// get the id of a specific planet and send it to the function
+navList.addEventListener("click", ({ target }) => {
+    const buttonId = target.parentElement.id;
+
+    let planetId = solarSystem.findIndex((item) => item.id == buttonId);
```
[the remark this answers is not in the document]
Comment (ASTkach) [reply submissions/ASTkach/js-dom/js/script.js:489]: Sorry, I didn't fully understand everything. Should I use the example above and rewrite my function?

File: submissions/ASTkach/js-dom/js/script.js [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+            planetTemplate += planetTemplateListEnd;
+            planetTemplate += planetTemplateEnd;
+
+            mainBlock.innerHTML = planetTemplate;
+        }
+    });
+}
+
+// create homepage
+getCurrentPlanet(0);
```
[the remark this answers is not in the document]
Comment (ASTkach) [reply submissions/ASTkach/js-dom/js/script.js:452]: my first "planet" id

File: submissions/ASTkach/js-dom/js/script.js (hunk 3 above)
Comment (ASTkach) [reply submissions/ASTkach/js-dom/js/script.js:366]: But i have some style on my "main__img" class.

File: submissions/ASTkach/js-dom/js/script.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (madmaxWMFU) [reply submissions/ASTkach/js-dom/js/script.js:452]: I understand but for other users, it is not informative.
```
const initialContentValue = 0;
getCurrentPlanet(initialContentValue);
```

File: submissions/ASTkach/js-dom/js/script.js (hunk 3 above)
Comment (madmaxWMFU) [reply submissions/ASTkach/js-dom/js/script.js:366]: Fine. Then these styles are redundant here.

File: submissions/ASTkach/js-dom/js/script.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (madmaxWMFU) [reply submissions/ASTkach/js-dom/js/script.js:489]: You look for the index of an element of the array by its id, and then go through the array and compare this index with the property of the element. That means you are doing double work.
And so you can use the ``Array.find()'' method to pass it an identifier by which you can find the necessary element of the array with information. And then, already selected information, you pass the function for display.

File: submissions/ASTkach/js-dom/js/script.js [hunk 6]
```diff
@@ -0,0 +1,495 @@
...
+    listMenu.classList.toggle("_opened");
+});
+
+// track the presence of the "_active" class for navigation elements
+// track the "click" event on them
+let navButton = document.getElementsByClassName("nav__button");
+let activeClass = document.getElementsByClassName("_active");
+for (i = 0; navButton.length > i; i++) {
+    navButton[i].onclick = function () {
+        let currentActive = activeClass[0];
```
Comment (madmaxWMFU) [inline submissions/ASTkach/js-dom/js/script.js:470]: Read about [event delegation](https://javascript.info/event-delegation)

Comment (ASTkach) [reply submissions/ASTkach/js-dom/js/script.js:470]: I tried to do it with event delegation. but my picture does not want to respond to a click in any way, unlike a paragraph. I even just removed all the styles from the page, but even that did not help

Comment (madmaxWMFU) [reply submissions/ASTkach/js-dom/js/script.js:470]: Find me in telegram '@madmaxWMFU'

**Review by madmaxWMFU — APPROVED** [review]
Great, I approve of your work. Good luck with future tasks!

**Pull request merged**